nomad/structs/structs.go

```diff
@@ -2074,6 +2074,7 @@ type ServiceCheck struct {
 	Interval      time.Duration // Interval of the check
 	Timeout       time.Duration // Timeout of the response from the check before consul fails the check
 	InitialStatus string        // Initial status of the check
+	TLSSkipVerify bool          // Skip TLS verification when Protocol=https
```

Comment:
I was going to implement it for non-agent checks in a followup PR (and doc it then), but that seems silly. Fixing.